src/app/icd/server/ICDManager.cpp

```diff
@@ -234,25 +236,35 @@ bool ICDManager::CheckInMessagesWouldBeSent()
             }
 
             // At least one registration would require a Check-In message
-            VerifyOrReturnValue(mSubInfoProvider->SubjectHasActiveSubscription(entry.fabricIndex, entry.monitoredSubject), true);
+            VerifyOrReturnValue(!RegistrationVerifier(entry.fabricIndex, entry.monitoredSubject), true);
         }
     }
 
     // None of the registrations would require a Check-In message
     return false;
 }
 
-void ICDManager::TriggerCheckInMessages()
+bool ICDManager::CheckInWouldBeSentAtActiveModeVerifier(FabricIndex aFabricIndex, NodeId subjectID)
```

Comment:
the function name is confusing here IMO

Reply:
We'll leave the name as is for now. Namming of the function was established in the PR that added the function and i would like to avoid reoppening that can of worms.